lib/Map/TableStructure.js

```diff
@@ -503,6 +514,121 @@ TableStructure.prototype.getIdMapping = function() {
     }, {});
 };
 
+/**
+ * Appends table2 to this table. If rowNumbers are provided, only takes those
+ * row numbers from table2.
+ * Changes all the columns in one go, to avoid partial updates from tracked values.
+ * @param {TableStructure} table2 The table to add to this one.
+ * @param {Integer[]} [rowNumbers] The row numbers from table2 to add (defaults to all).
+ */
+TableStructure.prototype.append = function(table2, rowNumbers) {
+    if (this.columns.length !== table2.columns.length) {
+        throw new DeveloperError('Cannot add tables with different numbers of columns.');
+    }
+    var updatedColumnValuesArrays = [];
+    function mapRowNumberToValue(valuesToAdd) {
+        return rowNumber => valuesToAdd[rowNumber];
+    }
+    for (var columnNumber = 0; columnNumber < table2.columns.length; columnNumber++) {
+        var valuesToAdd = table2.columns[columnNumber].values;
+        if (defined(rowNumbers)) {
+            valuesToAdd = rowNumbers.map(mapRowNumberToValue(valuesToAdd));
```

Comment:
that works too :). Although it means line 533 should probably be in an else block.